extensions/script-libraries/drawing/drawing.livecodescript

```diff
@@ -0,0 +1,2960 @@
+	put binaryEncode("I", the number of bytes in xContext["opcodes"] + the number of bytes in tExtraOpcodes) after tDrawing
+	put tExtraOpcodes after tDrawing
+	put xContext["opcodes"] after tDrawing
+	put tDrawing into rDrawing
+end _svgEncode
+
+private function _svgEncodeIndex pIndex
+	local tValue
+	put empty into tValue
+	repeat while pIndex >= 128
+		put numToByte(128 + (pIndex mod 128)) after tValue
+		put pIndex div 128 into pIndex
+	end repeat
+	put numToByte(pIndex mod 128) after tValue
+	return tValue
+end _svgEncodeIndex
+
+private command _svgEncodeOp @xContext, pOpcode
+	local tParams
+	repeat with i = 3 to the paramCount
+		put param(i) into tParams[i - 2]
+	end repeat
+	_svgEncodeOpV xContext, pOpcode, tParams
+end _svgEncodeOp
+
+private command _svgEncodeOpV @xContext, pOpcode, pArguments
+	put numToByte(pOpcode) after xContext["opcodes"]
+	repeat for each element tArgument in pArguments
+		if tArgument is an array or tArgument is empty then
+			put _svgEncodeIndex(the number of elements in tArgument) after xContext["opcodes"]
+			repeat for each element tElement in tArgument
+				if tElement is not a number then
+					_InternalError format("invalid scalar value '%s' in list", tArgument)
+				end if
+				put binaryEncode("f", tElement) after xContext["scalars"]
+			end repeat
+		else if tArgument is a number then
+			put binaryEncode("f", tArgument) after xContext["scalars"]
+		else
+			_InternalError format("invalid scalar value '%s'", tArgument)
+		end if
+	end repeat
+end _svgEncodeOpV
+
+private command _svgEncodePaint @xContext, pPaint
+	if pPaint[1] is "none" then
+	else if pPaint[1] is "color" then
+		_svgEncodeOp xContext, kMCGDrawingPaintOpcodeSolidColor, pPaint[2], pPaint[3], pPaint[4]
+	else
+		_InternalError format("unknown paint '%s'", pPaint[1])
+	end if
+	_svgEncodeOp xContext, kMCGDrawingPaintOpcodeEnd
+end _svgEncodePaint
+
+private function _svgEncodePreserveAspectRatio pPar
+	local tParOp
+	switch word 1 of pPar
+	case "none"
+		put kMCGDrawingPreserveAspectRatioOpcodeNone into tParOp
+		break
+	case "xMinYMin"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMinYMinMeet into tParOp
+		break
+	case "xMidYMin"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMidYMinMeet into tParOp
+		break
+	case "xMaxYMin"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMaxYMinMeet into tParOp
+		break
+	case "xMinYMid"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMinYMidMeet into tParOp
+		break
+	case "xMidYMid"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMidYMidMeet into tParOp
+		break
+	case "xMaxYMid"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMaxYMidMeet into tParOp
+		break
+	case "xMinYMax"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMinYMaxMeet into tParOp
+		break
+	case "xMidYMax"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMidYMaxMeet into tParOp
+		break
+	case "xMaxYMax"
+		put kMCGDrawingPreserveAspectRatioOpcodeXMaxYMaxMeet into tParOp
+		break
+	end switch
+	if word 2 of pPar is "slice" then
+		add 9 to tParOp
```

Comment:
Is it maybe worth putting an extra constant in for this 9? Not a big deal if the list of aspect ratio preservation types is never going to change I suppose.

Reply:
I can change it to compute based on the PAR constants - XminYminMeet + XminYminSlice ?

Reply:
Yeah, that works